newrelic/resource_newrelic_synthetics_monitor_test.go

```diff
@@ -0,0 +1,132 @@
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "uri", "https://google.com"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "locations.#", "1"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "locations.0", "US_EAST_1"),
+				),
+			},
+			{
+				Config: testAccCheckNewRelicSyntheticsMonitorConfigUpdated(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicSyntheticsMonitorExists("newrelic_synthetics_monitor.foo"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "name", fmt.Sprintf("%s-updated", rName)),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "type", "SIMPLE"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "frequency", "2"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "status", "ENABLED"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "uri", "https://github.com"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "locations.#", "2"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "locations.0", "US_EAST_1"),
+					resource.TestCheckResourceAttr(
+						"newrelic_synthetics_monitor.foo", "locations.1", "US_WEST_1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckNewRelicSyntheticsMonitorExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No synthetics monitor ID is set")
+		}
+
+		client := testAccProvider.Meta().(*ProviderConfig).Synthetics
+
+		found, err := client.GetMonitor(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if found.ID != rs.Primary.ID {
+			return fmt.Errorf("Synthetics monitor not found: %v - %v", rs.Primary.ID, found)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckNewRelicSyntheticsMonitorDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ProviderConfig).Synthetics
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "newrelic_synthetics_monitor" {
+			continue
+		}
+
+		_, err := client.GetMonitor(r.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Synthetics monitor still exists")
+		}
+
+	}
+	return nil
+}
+
+func testAccCheckNewRelicSyntheticsMonitorConfig(rName string) string {
+	return fmt.Sprintf(`
+
+resource "newrelic_synthetics_monitor" "foo" {
+  name = "%[1]s"
+  type = "SIMPLE"
+  frequency = 1
+  status = "DISABLED"
+  locations = ["AWS_US_EAST_1"]
+  uri = "https://google.com"
+}
+`, rName)
+}
+
+func testAccCheckNewRelicSyntheticsMonitorConfigUpdated(rName string) string {
+	return fmt.Sprintf(`
+
+resource "newrelic_synthetics_monitor" "foo" {
+  name = "%[1]s-updated"
+  type = "SIMPLE"
+  frequency = 2
```

Comment:
2 is not an allowed monitor frequency per New Relic API: https://docs.newrelic.com/docs/apis/synthetics-rest-api/monitor-examples/manage-synthetics-monitors-rest-api#simple-monitor-calls

This is breaking your acceptance tests